site/content/en/docs/Tasks/configmaps_and_secrets.md

```diff
@@ -1,8 +1,357 @@
 ---
-title: "ConfigMaps and Secrets"
-linkTitle: "ConfigMaps and Secrets"
-weight: 3
-date: 2017-01-05
+title: "Generating ConfigMaps and Secrets"
+linkTitle: "Generating ConfigMaps and Secrets"
+weight: 1
+date: 2023-10-06
 description: >
-  Working with ConfigMaps and Secrets
----
\ No newline at end of file
+    Working with ConfigMap and Secret Generators
+---
+
+ConfigMap and Secret objects can be generated by adding a `configMapGenerator` and `secretGenerator` entry to the `kustomization.yaml` file.
+
+## Create ConfigMap from a file
+
+ConfigMap Resources may be generated from files - such as a java `.properties` file.  To generate a ConfigMap Resource for a file, add an entry to `configMapGenerator` with the filename.
+
+The ConfigMaps will have data values populated from the file contents. The contents of each file will appear as a single data item in the ConfigMap keyed by the filename.
+
+The following example generates a ConfigMap with a data item containing the contents of a file.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: my-application-properties
+  files:
+  - application.properties
+```
+
+2. Create a `.properties` file.
+```yaml
+# application.properties
+FOO=Bar
+```
+
+3. Create the ConfigMap using `kustomize build`:
+```bash
+kustomize build .
```

Comment:
I'm not 100% sure, but shouldn't this be `.` instead of `:`? Same goes for all the build steps below.

```suggestion
3. Create the ConfigMap using `kustomize build`.
```bash
kustomize build .
```

I personally think that step 3 and 4 could be combined, but I guess your way is more specific.

Reply:
Replaced `:` with `.` and combined build and output steps.